modules/serverless-functions-func-yaml-fields.adoc

```diff
@@ -0,0 +1,137 @@
+[id="serverless-functions-func-yaml_{context}"]
+= Configurable fields in func.yaml
+
+Many of the fields in `func.yaml` are generated automatically when you create, build, and deploy your function. However, there are also fields that you modify manually to change things such as the function name or the image name.
+
+== builder
+
+The `builder` field specifies the Buildpack builder image to use when building the function. In most cases, this value should not be changed. When you do change it, use a value that is listed in the `builderMap` field.
+
+== builderMap
+
+Some function runtimes can be built in multiple ways. For example, a Quarkus function can be built for the JVM or as a native binary. The `builderMap` field contains all of the builders available for a given runtime.
+
+== envs
+
+The `envs` field enables you to set environment variables to be available to your function at runtime. You can set an environment variable in several different ways:
+
+. Directly from a value.
+. From a value assigned to a local environment variable. See the section "Referencing local environment variables from func.yaml fields" for more information.
+. From a key-value pair stored in a secret or config map.
+. You can also import all key-value pairs stored in a secret or config map, with keys used as names of the created environment variables.
+
+This examples demonstrates the different ways to set an environment variable:
+
+[source,yaml]
+----
+name: test
+namespace: ""
+runtime: go
+...
+envs:
+- name: EXAMPLE1 <1>
+  value: value
+- name: EXAMPLE2 <2>
+  value: '{{ env:LOCAL_ENV_VALUE }}'
+- name: EXAMPLE3 <3>
+  value: '{{ secret:mysecret:key }}'
+- name: EXAMPLE4 <4>
+  value: '{{ configMap:myconfigmap:key }}'
+- value: '{{ secret:mysecret2 }}' <5>
+- value: '{{ configMap:myconfigmap2 }}' <6>
+----
+<1> An environment variable set directly from a value.
+<2> An environment variable set from a value assigned to a local environment variable.
+<3> An environment variable assigned from a key-value pair stored in a secret.
+<4> An environment variable assigned from a key-value pair stored in a config map.
+<5> A set of environment variables imported from key-value pairs of a secret.
+<6> A set of environment variables imported from key-value pairs of a config map.
+
+== volumes
+
+The `volumes` field enables you to mount secrets and config maps as a volume accessible to the function at the specified path. For example:
```

Comment:
I've always tried to avoid fragments introducing an example, like "For example:" so this may be more of a suggestion. But I would suggestthis-- accessible to the function at the specified path, as shown in the following example:

Reply:
Got it, but I'll rather keep this - we have hundreds of instances of this wording in the docs.

Reply:
I'd actually prefer to see this updated as per @skrthomas 's suggestion if that's OK? You're not wrong that this probably exists in a bunch of other places, but having it in one less place is still an improvement 🙃

Reply:
@skrthomas @abrennan89 Changed as requested, but could you explain the reasoning behind this? Coz in my books it just made the text longer, plus when it's "For example:", it's a stock phrase, people know it and have less mental friction when reading & parsing. Why is your version preferred?

Reply:
I just think it reads better, it was just my personal opinion in response to what Sara said 🙂
There's no right or wrong answer to this I think, the IBM style guide has a section on _Code and command examples_ which has a bunch of ways to do this but like I said, I just prefer "...as shown in the following example" or similar.

Reply:
@abrennan89 "There's no right or wrong answer" "I just think it reads better" "just my personal opinion" "I just prefer" Love them peer reviews! :smile: 
And FWIW:
`$ git grep -E 'shown in the following example|shown in the example|shown in this example' | wc -l`
`25`

`$ git grep -E 'For example:' | wc -l`
`360`
:)